imcache_test.go

```diff
@@ -941,6 +942,53 @@ func TestImcache_GetAll_SlidingExpiration(t *testing.T) {
 	}
 }
 
+//GetAndRemove Tests Init
+func TestImcache_GetAndRemove(t *testing.T) {
```

Comment:
Could you also extend eviction policy tests (`TestCache_MaxEntriesLimit_EvictionPolicyLRU`, see https://github.com/erni27/imcache/blob/master/imcache_test.go#L1566, `TestCache_MaxEntriesLimit_EvictionPolicyLFU`, see https://github.com/erni27/imcache/blob/master/imcache_test.go#L1834 and `TestCache_MaxEntriesLimit_EvictionPolicyRandom`, see https://github.com/erni27/imcache/blob/master/imcache_test.go#L2085)?

Would be great to confirm that the new method doesn't really mess up with the underlying queue, you can take a look at how the `Remove` method is invoked in those tests are what are the expectations.